src/main/java/edu/kit/compiler/JavaEasyCompiler.java

```diff
@@ -385,23 +363,22 @@ public static void main(String[] args) {
         OptimizationLevel optimizationLevel = parseOptimizationLevel(cliCall);
         DebugFlags debugFlags = parseDebugFlags(cliCall);
 
-        // determine used optimizations
-        Iterable<Optimization> optimizations;
+        Optimizer optimizer;
         RegisterAllocator allocator;
         switch (optimizationLevel) {
             case Level0:
-                optimizations = Arrays.asList(
+                optimizer = new Optimizer(List.of(), List.of(
                     new ConstantOptimization(),
                     new ArithmeticIdentitiesOptimization()
-                );
+                ), debugFlags.isDumpGraphs());
                 allocator = new DumbAllocator();
                 break;
             case Level1:
-                optimizations = Arrays.asList(
+                optimizer = new Optimizer(List.of(), List.of(
                     new ConstantOptimization(),
                     new ArithmeticIdentitiesOptimization(),
                     new ArithmeticReplacementOptimization()
-                );
+                ), debugFlags.isDumpGraphs());
```

Comment:
I would prefer passing the entire `debugFlags` to the optimizer. That would make it easier to extend it, in case we want to use additional debug flags in the future. The usage in `Optimizer` would change only slightly to `debugFlags.isDumpGraphs`.